### PR TITLE
new sched: Priority and Lowest RTT first

### PR DESCRIPTION
**Pull request by matttbe**
This scheduler combines two attributes: priority and RTT. It looks like a good example because it shows the limits of combining attributes.

Note: there are two other small fixes prior that.

### CODE REVIEW

File: draft-schedulers.mkd [hunk 1]
```diff
@@ -494,6 +494,57 @@ window by 2.
...
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+class PriorityAndLowestRTTFirst(Scheduler):
+    """ Chooses the first available path with the highest priority and then the lowest RTT. """
+
+    def schedule(self, packet_len: int):
+        # Sort paths by ascending priority (2nd sort) and then ascending SRTT (1st sort)
+        paths = sorted(self.paths, key=lambda path: path.srtt)
+        paths = sorted(paths, key=lambda path: path.priority, reverse=True)
+        for p in sorted(self.paths, key=lambda path: path.srtt):
```
Comment (mpiraux) [inline draft-schedulers.mkd]: I guess you meant
```suggestion
        for p in paths:
```

File: scheduler_simulator.py [hunk 2]
```diff
@@ -445,7 +445,19 @@ class LowestRTTFirst(Scheduler):
...
+
+
+class PriorityAndLowestRTTFirst(Scheduler):
+    """ Chooses the first available path with the highest priority and then the lowest RTT. """
+
+    def schedule(self, packet_len: int) -> Optional[Path]:
+        # Sort paths by ascending priority (2nd sort) and then ascending SRTT (1st sort)
+        paths = sorted(self.paths, key=lambda path: path.srtt)
+        paths = sorted(paths, key=lambda path: path.priority, reverse=True)
         for p in sorted(self.paths, key=lambda path: path.srtt):
```
Comment (mpiraux) [inline scheduler_simulator.py]: I guess you also meant
```suggestion
        for p in paths:
```

**Review by mpiraux — COMMENTED** [review]
LGTM

File: scheduler_simulator.py (hunk 2 above)
Comment (matttbe) [reply scheduler_simulator.py]: Yes indeed, good catch!

**Pull request merged**